### PR TITLE
Accept reject invitations

### PR DESCRIPTION
**Pull request by maciekmichalik**

### CODE REVIEW

File: app/models/employee.rb [hunk 1]
```diff
@@ -2,4 +2,22 @@ class Employee < User
...
   has_many :employee_driving_schools
   has_many :driving_schools, through: :employee_driving_schools
+
+  # == Callbacks ==============================================================
+  after_create :find_pending_invitation
+
+  # == Instance Methods =======================================================
+  private
+
+  def find_pending_invitation
```
Comment (pospieszynski) [inline app/models/employee.rb]: please change function name to better depict its purpose, it does more than finding pending invitations.

File: app/models/employee.rb [hunk 2]
```diff
@@ -2,4 +2,22 @@ class Employee < User
...
+    invitations = Invitation.where('lower(email) = ?', self.email)
+
+    ActiveRecord::Base.transaction do
+      invitations.each do |invitation|
+        invitation.invitable.update(employee: self)
+      end
+
+      invitations.destroy_all
```
Comment (pospieszynski) [inline app/models/employee.rb]: I think we can use delete_all here, it;s faster and we dont want to remove associated record so it's ok I guess

File: app/models/student.rb [hunk 3]
```diff
@@ -2,4 +2,22 @@ class Student < User
...
   has_many :student_driving_schools
   has_many :driving_schools, through: :student_driving_schools
+
+  # == Callbacks ==============================================================
+  after_create :find_pending_invitation
+
+  # == Instance Methods =======================================================
+  private
+
+  def find_pending_invitation
```
Comment (pospieszynski) [inline app/models/student.rb]: the same situation with function name

File: spec/factories/invitations.rb [hunk 4]
```diff
@@ -1,8 +1,8 @@
 FactoryBot.define do
   factory :invitation do
     email "MyString"
-    first_name "MyString"
-    last_name "MyString"
+    name "MyString"
```
Comment (pospieszynski) [inline spec/factories/invitations.rb]: please utilize FFaker

File: spec/models/employee_spec.rb [hunk 5]
```diff
@@ -3,4 +3,25 @@
...
+      let!(:invitation_1) { create(:invitation, email: 'test@gmail.com', invitable: employee_driving_school_1) }
+      let!(:invitation_2) { create(:invitation, email: 'test@gmail.com', invitable: employee_driving_school_2) }
+
+      it 'is invoked after create' do
+        employee = build(:employee)
+        expect(employee).to receive(:find_pending_invitation)
+        employee.save
+      end
+
+      it 'finds pending invitations and assigns them to employee' do
```
Comment (pospieszynski) [inline spec/models/employee_spec.rb]: I think we don't assign invitation but newly created user to the school for which he was invited

File: spec/models/employee_spec.rb [hunk 6]
```diff
@@ -3,4 +3,25 @@
...
+    context '#find_pending_invitation' do
+      let(:employee_driving_school_1) { create(:employee_driving_school, status: :pending, employee: nil) }
+      let(:employee_driving_school_2) { create(:employee_driving_school, status: :pending, employee: nil) }
+
+      let!(:invitation_1) { create(:invitation, email: 'test@gmail.com', invitable: employee_driving_school_1) }
+      let!(:invitation_2) { create(:invitation, email: 'test@gmail.com', invitable: employee_driving_school_2) }
+
+      it 'is invoked after create' do
+        employee = build(:employee)
+        expect(employee).to receive(:find_pending_invitation)
```
Comment (pospieszynski) [inline spec/models/employee_spec.rb]: don't get how it works:[

File: config/routes.rb [hunk 7]
```diff
@@ -10,7 +10,12 @@
         member do
           put :confirm_registration
         end
-        resources :invitations, only: [:create]
+        resources :invitations, only: [:create] do
+          collection do
```
Comment (pospieszynski) [inline config/routes.rb]: shouldn't be a 'member' here? acceptance/rejection concerns only one relation at time, right?

Comment (maciekmichalik) [reply config/routes.rb]: it is under drving school scope

File: spec/models/employee_spec.rb (hunk 6 above)
Comment (maciekmichalik) [reply spec/models/employee_spec.rb]: I expect that after employee is being saved then it will receive method

**Pull request merged**